composer/callbacks/checkpoint_saver.py

```diff
@@ -301,119 +301,117 @@ def __init__(
-                                     artifact_name=artifact_name,
-                                     file_path=checkpoint_filepath,
-                                     overwrite=self.overwrite)
-
-            if self.latest_filename is not None:
-                formatted_folder_path = format_name_with_dist(self.folder, state.run_name)
-                symlink_name = os.path.join(
-                    formatted_folder_path,
-                    format_name_with_dist_and_time(
-                        self.latest_filename,
-                        state.run_name,
-                        state.timestamp,
-                    ).lstrip('/'),
-                )
-                if is_model_deepspeed(state.model) and not is_tar(symlink_name):
-                    # Deepspeed requires tarballs; appending `.tar`
-                    symlink_name += '.tar'
-                symlink_dirname = os.path.dirname(symlink_name)
-                if symlink_dirname:
-                    os.makedirs(symlink_dirname, exist_ok=True)
-                try:
-                    os.remove(symlink_name)
-                except FileNotFoundError:
-                    pass
-                relative_checkpoint_path = os.path.relpath(checkpoint_filepath, formatted_folder_path)
-                os.symlink(relative_checkpoint_path, symlink_name)
-                if self.artifact_name is not None and self.latest_artifact_name is not None:
-                    symlink_artifact_name = format_name_with_dist_and_time(self.latest_artifact_name, state.run_name,
-                                                                           state.timestamp).lstrip('/') + '.symlink'
-                    artifact_name = format_name_with_dist_and_time(self.artifact_name, state.run_name,
-                                                                   state.timestamp).lstrip('/')
-                    # Always overwrite for symlinks since we use the same filename for latest
-                    with tempfile.TemporaryDirectory() as tmpdir:
-                        symlink_filename = os.path.join(tmpdir, 'latest.symlink')
-                        create_symlink_file(artifact_name, symlink_filename)
-                        logger.file_artifact(
-                            log_level=log_level,
-                            artifact_name=symlink_artifact_name,
-                            file_path=symlink_filename,
-                            overwrite=True,
-                        )
-
-        timestamp = state.timestamp
-
-        self.saved_checkpoints.append((timestamp, checkpoint_filepaths))
+        is_deepspeed = is_model_deepspeed(state.model)
+
+        # save the checkpoint to the filename
+        filename = self.filename.format(state, is_deepspeed)
+
+        saved = checkpoint.save_checkpoint(
+            state=state,
+            filename=filename,
+            weights_only=self.weights_only,
+        )
+
+        if not saved:  # not all ranks save
+            return
```

Comment:
```suggestion
        saved_path = checkpoint.save_checkpoint(
            state=state,
            filename=filename,
            weights_only=self.weights_only,
        )

        if not saved_path:  # not all ranks save
            return
```
nit: `saved` sounds like bool